app/desktop/studio_server/tool_api.py

```diff
@@ -77,6 +131,52 @@ class ToolSetApiDescription(BaseModel):
     tools: list[ToolApiDescription]
 
 
+async def available_remote_mcp_tools(
+    server: ExternalToolServer,
+) -> list[ToolApiDescription]:
+    """
+    Get the available tools from a remote MCP server. If the server is not reachable, return an empty list.
+    """
+    try:
+        async with MCPSessionManager.shared().mcp_client(server) as session:
+            tools_result = await session.list_tools()
+            return [
+                ToolApiDescription(
+                    id=f"{MCP_REMOTE_TOOL_ID_PREFIX}{server.id}::{tool.name}",
+                    name=tool.name,
+                    description=tool.description,
+                )
+                for tool in tools_result.tools
+            ]
+    except Exception:
```

Comment:
generally don't ignore "Exception". If you want to ignore a specific exception that's fine, but keep it specific.